core/drivers/amd/gpio_common.c

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2024-2025, Advanced Micro Devices, Inc. All rights reserved.
+ *
+ */
+
+#include <assert.h>
+#include <gpio_private.h>
+
+#ifndef CFG_DRIVERS_GPIO
+#error AMD GPIO drivers expects CFG_DRIVERS_GPIO
+#endif
+
+void get_bank_and_pin(struct amd_gbank_data *bdata, uint32_t gpio,
+		      uint32_t *bank, uint32_t *pin)
+{
+	uint32_t i;
+
+	assert(gpio >= bdata->ngpio);
+
+	for (i = 0; i < bdata->max_bank; i++) {
+		if (gpio >= bdata->bank_min[i] &&
+		    gpio <= bdata->bank_max[i]) {
+			*bank = i;
+			*pin = gpio - bdata->bank_min[i];
+			break;
+		}
+	}
```

Comment:
If the GPIO is not found, the function returns with unmodified `*bank` and `*pin` values. I think panicking or returning an error code would be better.